exdb/views.py

```diff
@@ -46,52 +43,66 @@ def get_form_class(self):
             return ExperienceSaveForm
 
 
-class HallStaffDashboardView(TemplateView):
+class HomeView(ListView):
+    template_name = 'exdb/home.html'
     access_level = 'basic'
-    template_name = 'exdb/hallstaff_dash.html'
-
-    def get_context_data(self):
-        context = super(HallStaffDashboardView, self).get_context_data()
-        context['user'] = get_user_model().objects.prefetch_related(
-            'approval_queue__author',
-            'approval_queue__keywords',
-            'approval_queue__recognition',
-            'approval_set__experience__author',
-            'approval_set__experience__keywords',
-            'approval_set__experience__recognition'
-        ).get(pk=self.request.user.pk)
-        return context
+    context_object_name = 'experiences'
 
+    def is_hs(self):
+        return self.request.user.groups.filter(name='hs')
 
-class RAHomeView(ListView):
-    access_level = 'basic'
-    template_name = 'exdb/ra_home.html'
-    context_object_name = 'experiences'
+    def get_hs_queryset(self):
+        experience_approvals = ExperienceApproval.objects.filter(
+            approver=self.request.user, experience__status='ad'
+        )
+        experiences = Experience.objects.filter(
+            Q(next_approver=self.request.user) |
+            Q(pk__in=experience_approvals.values('experience'))
+        )
+        return experiences
+
+    def get_ra_queryset(self):
+        return Experience.objects.filter(Q(author=self.request.user) | (Q(planners=self.request.user) & ~Q(status='dr')))\
+            .exclude(status__in=('ca')).order_by('created_datetime').distinct()
 
     def get_queryset(self):
-        return Experience.objects.filter(author=self.request.user).order_by('created_datetime')
+        if self.is_hs():
+            return self.get_hs_queryset()
+        else:
+            return self.get_ra_queryset()
 
     def get_context_data(self, *args, **kwargs):
-        context = super(RAHomeView, self).get_context_data(*args, **kwargs)
-        context['ra'] = self.request.user
+        context = super(HomeView, self).get_context_data(*args, **kwargs)
+        context['user'] = self.request.user
+
+        # This is what experience groups we are showing the user and in what order
+        if self.is_hs():
+            status_to_display = [_('Pending Approval'), _('Approved')]
+        else:
+            status_to_display = [_(x[1]) for x in Experience.STATUS_TYPES]
```

Comment:
For non hall staff users, the status order should be: `[_('Needs Evaluation')] + [_(x[1]) for x in Experience.STATUS_TYPES]`.